### PR TITLE
Feature/vih 5544 add stryker mutation testing

### PR DESCRIPTION
**Pull request by shravankris**
https://tools.hmcts.net/jira/browse/VIH-5544

- Added Stryker mutation tool and settings config
- Added test coverages and fixed failing mutants on unit tests

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: UserApi/Testing.Common/TestConfig.cs [hunk 1]
```diff
@@ -12,7 +12,7 @@ private TestConfig()
         {
             var configRootBuilder = new ConfigurationBuilder()
                 .AddJsonFile("appsettings.json")
-                .AddUserSecrets<Startup>();
+                .AddUserSecrets("CF55F1BB-0EE3-456A-A566-70E56AC24C95");
```
Comment (RuwanikaWeerakkody) [inline UserApi/Testing.Common/TestConfig.cs]: Hi Sarav, Why is this secret hardcoded? Should not expose this one.

**Review by RuwanikaWeerakkody — CHANGES_REQUESTED** [review]

File: UserApi/Testing.Common/TestConfig.cs (hunk 1 above)
Comment (shravankris) [reply UserApi/Testing.Common/TestConfig.cs]: @RuwanikaWeerakkody encountered an issue with one of the services unit test requiring settings, was failing and lacking UserSecretId. I tried few work around but was helpless. Probably I can try passing the object manually to see if it helps.

Comment (shravankris) [reply UserApi/Testing.Common/TestConfig.cs]: @RuwanikaWeerakkody  This has been reverted now. Thanks!

File: UserApi/UserApi.UnitTests/Services/UserAccountService/UserAccountServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+using UserApi.Helper;
+using UserApi.Security;
+using UserApi.Services;
+using UserApi.Services.Models;
+
+namespace UserApi.UnitTests.Services
+{
+    public class UserAccountServiceTests
+    {
+        protected const string Domain = "@hearings.reform.hmcts.net";
```
Comment (npoxon) [inline UserApi/UserApi.UnitTests/Services/UserAccountService/UserAccountServiceTests.cs]: make this a dummy or reference the secrets

**Review by npoxon — CHANGES_REQUESTED** [review]

**Review by npoxon — APPROVED** [review]

**Review by RuwanikaWeerakkody — APPROVED** [review]

**Pull request merged**